### PR TITLE
boost: Add 1.80.0

### LINKED ISSUE #12564: [request] boost/1.80.0
### Package Details
  * Package Name/Version: **boost/1.80.0**
  * Changelog: **https://www.boost.org/users/history/version_1_80_0.html**

The above mentioned version is newly released by the upstream project and not yet available as a recipe. Please add this version.

@jwillikers @datalogics-robb 
I saw you two were the last contributors
I'd be very grateful for this one

### PR DESCRIPTION
**Pull request by jwillikers**
Specify library name and version:  **boost/1.80.0**

Fixes #12564.

---

- [X] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [X] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [X] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [X] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by jwillikers** [comment]
I can't reproduce the `libiconv` issues on Linux and the logs here don't provide enough information for me to diagnose the problem. Unless  someone can spot something wrong with the related patch I had to update, see [here](https://github.com/conan-io/conan-center-index/pull/12807/files#diff-a1f0c325abfbf82c4e7d0df08fca1267cc98d0cf9376362813be62c7e68ff50c), then someone with  macOS will need to provide some support to get this fixed.

**Comment by TheSalvator** [comment]
> I can't reproduce the `libiconv` issues on Linux and the logs here don't provide enough information for me to diagnose the problem. Unless someone can spot something wrong with the related patch I had to update, see [here](https://github.com/conan-io/conan-center-index/pull/12807/files#diff-a1f0c325abfbf82c4e7d0df08fca1267cc98d0cf9376362813be62c7e68ff50c), then someone with macOS will need to provide some support to get this fixed.

I have macOS. I think I can test it after 4-5 hours (after work)

**Comment by TheSalvator** [comment]
I decided to try to build boost with -d9 (9th level of debug) b2 option.

What a mistake

**Comment by jwillikers** [comment]
> I think i fixed the issue, but i don't understand how to contribute a commit. I tried to push a branch into jwillikers:conan-center-index but got rejected.

You wouldn't have write permission to my repository. I don't have much experience with doing multi-contributor PR's, but I bet you can fork `conan-center-index` and create a PR to my fork's branch. I can then merge in your changes.

**Comment by TheSalvator** [comment]
> > I think i fixed the issue, but i don't understand how to contribute a commit. I tried to push a branch into jwillikers:conan-center-index but got rejected.
> 
> You wouldn't have write permission to my repository. I don't have much experience with doing multi-contributor PR's, but I bet you can fork `conan-center-index` and create a PR to my fork's branch. I can then merge in your changes.

Here it is
https://github.com/jwillikers/conan-center-index/pull/5

**Comment by TheSalvator** [comment]
C'mon. I have already been a contributor. I think i messed up user.name in commit(

**Comment by jwillikers** [comment]
> > > I think i fixed the issue, but i don't understand how to contribute a commit. I tried to push a branch into jwillikers:conan-center-index but got rejected.
> > 
> > 
> > You wouldn't have write permission to my repository. I don't have much experience with doing multi-contributor PR's, but I bet you can fork `conan-center-index` and create a PR to my fork's branch. I can then merge in your changes.
> 
> Here it is [jwillikers#5](https://github.com/jwillikers/conan-center-index/pull/5)

@TheSalvator, thanks for figuring out the macOS specific fixes necessary to get this working! I've incorporated your changes in my PR, but you'll need to make sure to sign Conan Center's CLA [here](https://cla-assistant.io/conan-io/conan-center-index?pullRequest=12807). Thanks again!

**Comment by TheSalvator** [comment]
> > > > I think i fixed the issue, but i don't understand how to contribute a commit. I tried to push a branch into jwillikers:conan-center-index but got rejected.
> > > 
> > > 
> > > You wouldn't have write permission to my repository. I don't have much experience with doing multi-contributor PR's, but I bet you can fork `conan-center-index` and create a PR to my fork's branch. I can then merge in your changes.
> > 
> > 
> > Here it is [jwillikers#5](https://github.com/jwillikers/conan-center-index/pull/5)
> 
> @TheSalvator, thanks for figuring out the macOS specific fixes necessary to get this working! I've incorporated your changes in my PR, but you'll need to make sure to sign Conan Center's CLA [here](https://cla-assistant.io/conan-io/conan-center-index?pullRequest=12807). Thanks again!

I messed up again. A already have signed CLA. Can you reset my PR and force push so I change the commit author and reopen the PR

**Comment by jwillikers** [comment]
@TheSalvator Done.

**Comment by TheSalvator** [comment]
@jwillikers Here you go)
https://github.com/jwillikers/conan-center-index/pull/6

**Comment by TheSalvator** [comment]
@jwillikers It's a pleasure working with you. Hope there is no more patches left to submit)

**Comment by jwillikers** [comment]
> @jwillikers It's a pleasure working with you. Hope there is no more patches left to submit)

Same!

**Comment by uilianries** [comment]
> @uilianries What is `continuous-integration/jenkins/pr-merge`? It takes suspiciously long to complete

@TheSalvator it's the CI build, it will show the final result. It seems be supicious, but Boost takes almost an entire day to be built. One thing is building locally, another thing is generating more than 100 builds for each version.

Both Boost and Qt are our Nemesis in terms of CI, they take both several hours to be built, and for each new commit we need to build all again.

**Comment by gegles** [comment]
ANy update on this? Are we sure it's still running?

**Comment by prince-chrismc** [comment]
I think we should drop some less popular versions... this way it wont overwhelm CCI so much.
Using my magic goggle to see into the backend, 1.70 and 1.73 are the least popular. I think we should drop those from the build 🔫 at least

https://github.com/conan-io/conan-center-index/blob/master/docs/faqs.md#what-is-the-policy-for-removing-older-versions-of-a-package

I pinged the CCI team to see if it's stuck, two days is a long time.

File: recipes/boost/all/conandata.yml [hunk 1]
```diff
@@ -240,5 +246,33 @@ patches:
...
+      patch_type: "conan"
+      base_path: "source_subfolder"
+    - patch_file: "patches/boost_1_77_mpi_check.patch"
+      patch_description: "Fails the build when mpi is not configured"
+      patch_type: "conan"
+      base_path: "source_subfolder"
+    - patch_file: "patches/1.80.0-0001-filesystem-win-fix-dir-it-net-share.patch"
+      patch_description: "Directory iterators may fail to construct for a network share on Windows prior to 10"
+      patch_type: "official"
+      patch_source: "https://github.com/boostorg/filesystem/issues/245"
```
Comment (prince-chrismc) [inline recipes/boost/all/conandata.yml:262]: I absolutely appreciate that someone took the time to add these link ❤️

**Review by prince-chrismc — APPROVED** [review]

**Comment by puremourning** [comment]
If it's useful in terms of testing, I have successfully exported this recipe and built it debug and release on both intel and arm Macs, with and without ICU

**Review by gegles — APPROVED** [review]

**Review by uilianries — APPROVED** [review]

File: recipes/boost/all/conandata.yml (hunk 1 above)
Comment (jwillikers) [reply recipes/boost/all/conandata.yml:262]: Thanks! They are super helpful for context and keeping track of what needs to be pulled forward.

**Review by jgsogo — APPROVED** [review]
The more people approve, the more resources assigned by the CI 😄  (it would be a nice feature to have)

**Comment by SpaceIm** [comment]
Are you sure it's not stuck? Usually boost takes few hours in CI, not days.

**Comment by SpaceIm** [comment]
@jgsogo https://github.com/conan-io/conan-center-index/pull/12795#issuecomment-1245161557

**Pull request merged**